ibm/data_source_ibm_database.go

```diff
@@ -0,0 +1,546 @@
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			"resource_group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The id of the resource group in which the Database instance is present",
+			},
+
+			"location": {
+				Description: "The location or the environment in which Database instance exists",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+
+			"service": {
+				Description: "The name of the Cloud Internet database service",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"plan": {
+				Description: "The plan type of the Database instance",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"status": {
+				Description: "The resource instance status",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"adminuser": {
+				Description: "The admin user id for the instance",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"adminpassword": {
+				Description: "The admin user id for the instance",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+			},
+			"version": {
+				Description: "The database version to provision if specified",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"members_memory_allocation_mb": {
+				Description: "Memory allocation required for cluster",
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+			},
+			"members_disk_allocation_mb": {
+				Description: "Disk allocation required for cluster",
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+			},
+			"users": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Description: "User name",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"password": {
+							Description: "User password",
+							Type:        schema.TypeString,
+							Optional:    true,
+							Sensitive:   true,
+						},
+					},
+				},
+			},
+			"connectionstrings": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Description: "User name",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"string": {
+							Description: "Connection string",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"certname": {
+							Description: "Certificate Name",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"certbase64": {
+							Description: "Certificate in base64 encoding",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"whitelist": {
+				Type:     schema.TypeSet,
+				Optional: true,
```

Comment:
In datasource whitelist is optional or computed..?.From code I see we are fetching from API and set so it can be just computed

Reply:
Agreed. These are all optional. Likely cut and paste copy from the resource definition.